File: Exercicio_17/biblioteca/src/entities/Livro.java
package entities;

public class Livro {
	
	protected String titulo;
	protected String autor;
	protected int anoPublicacao;
	protected boolean disponivel;
	
	public Livro(String titulo, String autor, int anoPublicacao) {
		this.titulo = titulo;
		this.autor = autor;
		this.anoPublicacao = anoPublicacao;
		this.disponivel = true;
	}
	
	public String getTitulo() {
		return titulo;
	}
	
	public void setTitulo(String titulo) {
		this.titulo = titulo;
	}
	
	public int getAnoPublicacao() {
		return anoPublicacao;
	}
	
	public void emprestar() {
		this.disponivel = false;
	}

	public void devolver() {
		this.disponivel = true;
	}
	
	public void exibirInfo() {
		System.out.println();
		System.out.println("Título: " + titulo);
		System.out.println("Autor: " + autor);
		System.out.println("Ano de Publicação: " + anoPublicacao);
		System.out.println("Disponível: " + (disponivel ? "Sim" : "Não"));
	}

}

File: 17_OrientacaoAObjetos/Livro.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace _17_OrientacaoAObjetos
{
    internal class Livro
    {

        public string Titulo { get; set; }
        public string Autor { get; set; }
        public int AnoPublicacao { get; set; }
        public bool Disponivel { get; set; } = true;

        public void Emprestar()
        {
            if (Disponivel)
            {
                Disponivel = false;
                Console.WriteLine("Livro emprestado com sucesso!");
            }
            else
            {
                Console.WriteLine("Livro já está emprestado!");
            }
        }

        public void Devolver()
        {
            Disponivel = true;
            Console.WriteLine("Livro devolvido.");
        }

        public virtual void ExibirInfo() 
        {
            Console.WriteLine("\nTítulo: " + Titulo);
            Console.WriteLine("Autor: " + Autor);
            Console.WriteLine("Ano: " + AnoPublicacao);
            Console.WriteLine("Disponível: " + Disponivel);
            Console.WriteLine();
        }


    }
}
